### PR TITLE
Add a Pkg3-style Project file

### PR DESCRIPTION
**Pull request by ararslan**

### CODE REVIEW

**Comment by nalimilan** [comment]
I thought we shouldn't add Project files for now? But I'm not up to date regarding these issues.

**Comment by ararslan** [comment]
Should be fine since this is mostly a leaf package and its only dependency, Compat, also has a Project file. The problem is adding a Project file to a package whose dependencies do not have them.

**Review by nalimilan — APPROVED** [review]

File: .travis.yml [hunk 1]
```diff
@@ -17,17 +18,21 @@ git:
...
-#  - julia -e 'Pkg.clone(pwd()); Pkg.build("Nullables"); Pkg.test("Nullables"; coverage=true)'
+script:
+  - if [[ -a .git/shallow ]]; then git fetch --unshallow; fi
+  - |
+      julia -e '
+        if VERSION >= v"0.7.0-DEV.3656"
+            using Pkg
+        end
+        if VERSION >= v"0.7.0-DEV.5183"
+            Pkg.develop(pwd())
```
Comment (ararslan) [inline .travis.yml:29]: This is apparently incorrect

**Comment by andreasnoack** [comment]
We'll revisit once we drop METADATA

**Comment by ararslan** [comment]
Hopefully by then this package can just die out.

**Pull request closed without merging**